core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -106,7 +106,22 @@ private[spark] abstract class Task[T](val stageId: Int, var partitionId: Int) ex
     if (interruptThread && taskThread != null) {
       taskThread.interrupt()
     }
-  }  
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq(stageId, partitionId)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
```

Comment:
This seems like an excessively complex way of writing `31 * stageId.hashCode + partitionId.hashCode`. I don't think FP is the way to do this.